mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -69,6 +68,14 @@
 
 public class StoreMessageIdManager implements MessageIdManager {
     private static final Logger LOGGER = LoggerFactory.getLogger(StoreMessageIdManager.class);
+    public static <T, S> Predicate<T> compose(Predicate<S> predicate, Function<T, S> function) {
+        return input -> predicate.test(function.apply(input));
+    }
+
+    static<T> Predicate<T> not(Predicate<T> p) {
+        return t -> !p.test(t);
+    }
```

Comment:
these 2 methods should be in java8-utils, no?

Reply:
Removed, we can use .negate() directly